infra/gcp/ensure-e2e-projects.sh

```diff
@@ -0,0 +1,129 @@
+# testing of kubernetes and managed by boskos
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+SCRIPT_DIR=$(dirname "${BASH_SOURCE[0]}")
+. "${SCRIPT_DIR}/lib.sh"
+
+function usage() {
+    echo "usage: $0 [repo...]" > /dev/stderr
+    echo "example:" > /dev/stderr
+    echo "  $0 # do all projects" > /dev/stderr
+    echo "  $0 k8s-infra-node-e2e-project # just do one" > /dev/stderr
+    echo > /dev/stderr
+}
+
+## setup service accounts and ips for the prow build cluster
+
+# TODO: replace prow-build-test with actual service account
+PROW_BUILD_SVCACCT=$(svc_acct_email "kubernetes-public" "prow-build-test")
+ensure_service_account \
+  "kubernetes-public" \
+  "prow-build-test" \
+  "used by prowjobs that run in prow-build-test cluster"
+# the namespace "test-pods" here must match the namespace defined in prow's config.yaml
+# to launch pods defined by prowjobs
+# eg: https://github.com/kubernetes/test-infra/blob/master/config/prow/config.yaml#L73
+empower_ksa_to_svcacct \
+  "kubernetes-public.svc.id.goog[test-pods/prow-build]" \
+  "kubernetes-public" \
+  "${PROW_BUILD_SVCACCT}"
+
+# manual parts: 
+# - create key, add to prow-build-test as service-account secret
+# - gsutil iam ch serviceAccount:$PROW_BUILD_SVCACCT:objectAdmin gs://bashfire-prow
+# - gsutil iam ch serviceAccount:$PROW_BUILD_SVCACCT:objectCreator gs://bashfire-prow
+
+# TODO: replace boskos-janitor-test with actual service account
+BOSKOS_JANITOR_SVCACCT=$(svc_acct_email "kubernetes-public" "boskos-janitor-test")
+ensure_service_account \
+  "kubernetes-public" \
+  "boskos-janitor-test" \
+  "used by boskos-janitor in prow-build-test cluster"
+# the namespace "test-pods" here must match the namespace defined in prows config.yaml
+# to launch pods defined by prowjobs because most prowjobs as-written assume they can
+# talk to either http://boskos (kubetest or bootstrap.py jobs) or 
+# https://boskos.svc.test-pods.cluster.local (some of the cluster-api jobs), and so
+# all boskos components are deployed to this namespace
+empower_ksa_to_svcacct \
+  "kubernetes-public.svc.id.goog[test-pods/boskos-janitor]" \
+  "kubernetes-public" \
+  "${BOSKOS_JANITOR_SVCACCT}"
+ensure_regional_address \
+  "kubernetes-public" \
+  "us-central1" \
+  "boskos-metrics" \
+  "to allow monitoring.k8s.prow.io to scrape boskos metrics"
+
+## setup projects to be used by e2e tests for standing up clusters
+
+# TODO: replace spiffxp- projects with actual projects
+E2E_PROJECTS=(
+  # for manual use during node-e2e job migration, eg: --gcp-project=spiffxp-node-e2e-project
+  spiffxp-node-e2e-project
+  # for manual use during job migration, eg: --gcp-project=spiffxp-gce-project
+  spiffxp-gce-project
+  # managed by boskos, part of the gce-project pool, eg: --gcp-project-type=gce-project
+  spiffxp-boskos-project-01
+  spiffxp-boskos-project-02
+  spiffxp-boskos-project-03
+)
+
+if [ $# = 0 ]; then
+    # default to all e2e projects
+    set -- "${E2E_PROJECTS[@]}"
+fi
+
+for prj; do
```

Comment:
The other ensure-* scripts all use `color` and `indent` to make the output more human-friendly.  Any reason not to do that here?

Reply:
It was mostly laziness.  Turns out indent didn't work with macOS' default version of sed, so now it does. The `color 6` thing doesn't actually color output on my terminal but I'm not going to troubleshoot that today.